plugins/metrics/mana.go

```diff
@@ -6,16 +6,26 @@ import (
 	"github.com/iotaledger/goshimmer/packages/mana"
 	"github.com/iotaledger/goshimmer/plugins/autopeering/local"
 	manaPlugin "github.com/iotaledger/goshimmer/plugins/mana"
+	"github.com/iotaledger/hive.go/identity"
 	"go.uber.org/atomic"
 )
 
+// NodePledgeMap is a map of node and a list of mana pledges.
+type NodePledgeMap map[identity.ID][]float64
+
 var (
-	accessMap           mana.NodeMap
-	accessPercentile    atomic.Float64
-	accessLock          sync.RWMutex
-	consensusMap        mana.NodeMap
-	consensusPercentile atomic.Float64
-	consensusLock       sync.RWMutex
+	accessMap                 mana.NodeMap
+	accessPercentile          atomic.Float64
+	accessLock                sync.RWMutex
+	consensusMap              mana.NodeMap
+	consensusPercentile       atomic.Float64
+	consensusLock             sync.RWMutex
+	accessPledgeLock          sync.RWMutex
+	accessPledge              = map[uint][]float64{}
```

Comment:
currently we only log the pledges of our own node, but have the ability to do so for all nodes in the network.
```go
accessPledge = NodePledgeMap{}

type NodePledgeMap map[identity.id]PledgeLog

type PledgeLog struct {
    BM1Pledges []float64
    BM2Pledges []float64
}
```
and then we could define the methods on these types.
```go
func (p PledgeLog) AddBM1(val float54) {
    p.BM1Pledged = append(p.BM1Pledged, val)
}

func (p PledgeLog) GetBM1Average() float64 {
    for _, val := range p.BM1Pledges {...}
    return avg
}
```
.. and so on. Not only it makes the code more readable, but we measure average pledge amounts of all nodes.